Source/LibationSearchEngine/BookRule.cs

```diff
@@ -0,0 +1,21 @@
+﻿using DataLayer;
```

Comment:
There's certainly nothing wrong with having a BookRule. I'm not clear on what it gains you. Since every LibraryBook has `.Book`

Reply:
I really didn't want to make it. This whole refactor would have been so much cleaner and more beautiful if I could have just used library book everywhere. The reason I didn't is because of how user-defined items are updated. Those methods take a Book, not a LibraryBook. I started looking up the stack to see if I could change all callers to use library book instead, but a lot of stuff would need to be changed. I decided the lesser of two evils was to do what I did here.

Reply:
Makes sense

Reply:
We've talked about making such a change in various issues, and it probably won't happen, but if it does I think it will be a Many LibraryBook to One Book, so that would still be fine.

Reply:
LibraryBook is an ownership record of 1 book .: anything you can do to a book/user defined item you can do via LibraryBook. If we allow for duplicate books, it just won't be 1:1 in the other direction: a book could be owned by multiple library books; a library book still only owns 1 book. Even *if* we one day do expand to this (and 7 years later, this is nothing I've pursued for long), it would be trivial to have:

`void UpdateUserDefinedItem(this LibraryBook lb, var foo) => lb.Book.UpdateUserDefinedItem(foo)`

Reply:
> We've talked about making such a change in various issues, and it probably won't happen, but if it does I think it will be a Many LibraryBook to One Book, so that would still be fine.

Yup, looks like we commented with the same idea at the same time. Great minds and all that